x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/process/AnalyticsProcessManager.java

```diff
@@ -66,13 +70,18 @@ private void processData(String jobId, DataFrameDataExtractor dataExtractor, Ana
             LOGGER.info("[{}] Result processor has completed", jobId);
         } catch (IOException e) {
             LOGGER.error(new ParameterizedMessage("[{}] Error writing data to the process", jobId), e);
+            finishHandler.accept(e);
```

Comment:
looks to me like finishHandler could be called twice, here and as part of the finally block. I guess this isn't the way you want it.

Reply:
Good spot. This is actually where we should be setting the state to `FAILED` probably. I removed the call to the `finishHandler` and added a TODO to handle this failure.